x-pack/plugins/ml/public/ui_actions/open_in_anomaly_explorer_action.tsx

```diff
@@ -5,30 +5,109 @@
  * 2.0.
  */
 
+import type { TimeRange } from '@kbn/es-query';
 import { i18n } from '@kbn/i18n';
-import type { SerializableRecord } from '@kbn/utility-types';
-import type { UiActionsActionDefinition } from '@kbn/ui-actions-plugin/public';
+import type { MlEntityField } from '@kbn/ml-anomaly-utils';
 import { ML_ENTITY_FIELD_OPERATIONS } from '@kbn/ml-anomaly-utils';
-import type { MlCoreSetup } from '../plugin';
+import { isPopulatedObject } from '@kbn/ml-is-populated-object';
+import type {
+  EmbeddableApiContext,
+  HasParentApi,
+  HasType,
+  PublishesUnifiedSearch,
+  PublishingSubject,
+} from '@kbn/presentation-publishing';
+import { apiHasType, apiIsOfType } from '@kbn/presentation-publishing';
+import { createAction } from '@kbn/ui-actions-plugin/public';
+import type { SerializableRecord } from '@kbn/utility-types';
 import { ML_APP_LOCATOR } from '../../common/constants/locator';
-import type { AnomalyChartsFieldSelectionContext, SwimLaneDrilldownContext } from '../embeddables';
+import type { ExplorerAppState } from '../../common/types/locator';
+import type { AppStateSelectedCells } from '../application/explorer/explorer_utils';
+import type {
+  AnomalyExplorerChartsEmbeddableType,
+  AnomalySwimLaneEmbeddableType,
+} from '../embeddables';
 import {
   ANOMALY_EXPLORER_CHARTS_EMBEDDABLE_TYPE,
   ANOMALY_SWIMLANE_EMBEDDABLE_TYPE,
-  isAnomalyExplorerEmbeddable,
-  isSwimLaneEmbeddable,
 } from '../embeddables';
-import type { ExplorerAppState } from '../../common/types/locator';
+import type { MlCoreSetup } from '../plugin';
+import type { JobId } from '../shared';
+
+export interface AnomalyChartsFieldSelectionApi {
+  // Props from embeddable output
+  entityFields: PublishingSubject<MlEntityField[] | undefined>;
+}
+
+export interface SwimLaneDrilldownApi {
+  // Props from embeddable input
```

Comment:
Do we need these comments? Ideally soon these won't have any connection to the legacy embeddable's input / output.

Reply:
Did the cleanup in https://github.com/elastic/kibana/pull/178729/commits/22df822cd4f95d8a9f4eea8b2bf3233bf268045d